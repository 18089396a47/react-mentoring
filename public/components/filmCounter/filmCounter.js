import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

const FilmCounter = (props) => (
    <span>{props.count > 0 ? `${props.count} movies found` : null}</span>
);

FilmCounter.propTypes = {
    count: PropTypes.number
};

const mapStateToProps = (state) => {
    return {
        count: state.films.data.results.length
    };
};

export default connect(mapStateToProps)(FilmCounter);
