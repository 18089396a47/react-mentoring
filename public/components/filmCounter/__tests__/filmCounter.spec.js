import React from 'react';
import renderer from 'react-test-renderer';
import { FilmCounter } from '../filmCounter';


describe('FilmCounter component', () => {
    it('FilmCounter should show empty state', () => {
        const tree = renderer.create(
            <FilmCounter count={0} />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('FilmCounter should show film count', () => {
        const tree = renderer.create(
            <FilmCounter count={1} />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
