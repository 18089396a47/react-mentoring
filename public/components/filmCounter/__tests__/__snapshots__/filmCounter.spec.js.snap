// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmCounter component FilmCounter should show empty state 1`] = `<span />`;

exports[`FilmCounter component FilmCounter should show film count 1`] = `
<span>
  1 movies found
</span>
`;
